### PR TITLE
Add js client docs

### PR DESCRIPTION
**Pull request by adamsilverstein**
- initial pass based on client-js repo readme

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@adamsilverstein assigning this to you until you think it's ready to go.

**Comment by adamsilverstein** [comment]
@danielbachhuber Ok, I will keep reviewing and refining it, lets make sure to merge before you release the next API beta; I will remind you.

File: extending/javascript-client/index.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+Activate the WP-API plugin. Enqueue the script directly:
+
+```php
+wp_enqueue_script( 'wp-api' );
+```
+
+or as a dependency for your script:
+
+```php
+wp_enqueue_script( 'my_script', 'path_to_my_script', array( 'wp-api' ) );
```
Comment (kadamwhite) [inline extending/javascript-client/index.md]: this is a total nitpick but I generally prefer `path/to/my/script`, to re-emphasize that the string corresponds to a system path; the underscores look like a variable.

File: extending/javascript-client/index.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+```
+
+The library parses the root endpoint (the 'Schema') and creates matching Backbone models and collections. You will now have two root objects available to you: `wp.api.models` and `wp-api.collections`.
```
Comment (kadamwhite) [inline extending/javascript-client/index.md]: `wp.api.models` but `wp_api.collections`? Underscore looks like a typo, either way it's inconsistent

File: extending/javascript-client/index.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+These objects contain the following:
```
Comment (kadamwhite) [inline extending/javascript-client/index.md]: slightly ambiguous; might add "constructors" after "following"

File: extending/javascript-client/index.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+ * Posts
+ * Revisions
+ * Statuses
+ * Tags
+ * Taxonomies
+ * Types
+ * Users
+```
+
+You can use these endpoints as is to read, update, create and delete items using standard Backbone methods (fetch, sync, save & destroy for models, sync for collections). You can also extend these objects to make them your own, and build your views on top of them.
```
Comment (kadamwhite) [inline extending/javascript-client/index.md]: > as-is

File: extending/javascript-client/index.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+Each model and collection contains a list of methods the corrosponding endpoint supports. For example, models created from `wp.api.models.Posts` have a method array of:
+
+```
+["GET", "POST", "PUT", "PATCH", "DELETE"]
+```
+
+### Accepted options
+
+Each model and collection contains a list of options the corrosponding endpoint accepts (passed as a second parameter), for example:
```
Comment (kadamwhite) [inline extending/javascript-client/index.md]: > corresponding

The purpose of the "passed as a second parameter" clause is ambiguous, I think a reader would be confused where you're passing that parameter

File: extending/javascript-client/index.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+// Create a new post
+var post = new wp.api.models.Posts({ title:'new test' } );
+post.save();
+
+// Get a collection of the post's categories
+var postCategories = post.getCategories();
+
+// The new post has an single Category: Uncategorized
+postCategories.at( 0 ).get('name');
```
Comment (kadamwhite) [inline extending/javascript-client/index.md]: Also a nit but the JS in this block has inconsistent quote mark/spacing code style

File: extending/javascript-client/index.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+var appleCategory = allCategories.findWhere( { slug: 'apples' } );
+
+// Add the category to the postCategories collection
+appleCategory.set( 'parent_post', post.get( 'id' ) );
+postCategories.create( appleCategory.toJSON(), { type: 'POST' } );
+
+// Remove the Uncategorized category
+postCategories.at( 0 ).destroy();
+
+// Check the results - refectch
```
Comment (kadamwhite) [inline extending/javascript-client/index.md]: > re-fetch

File: extending/javascript-client/index.md [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+postsCollection.fetch( { data: { 'filter': { 'orderby': 'title', 'order': 'ASC' } } } );
+```
+
+All collections support pagination automatically, and you can get the next page of results using `more`:
+
+```
+postsCollection.more();
+```
+
+If you add custom endpoints to the api they will also become available as models/collections. For example, you will get new models and collections when you [add REST API support to your custom post type](http://v2.wp-api.org/extending/custom-content-types/). Note that you may need to open a new tab to get a new read of the Schema.
```
Comment (kadamwhite) [inline extending/javascript-client/index.md]: Not sure what the new tab is meant to accomplish -- is the issue browser caching?

**Comment by kadamwhite** [comment]
@adamsilverstein Read through this with a critical eye, but so glad that you're adding this! It'll be a big help.

File: extending/javascript-client/index.md (hunk 1 above)
Comment (adamsilverstein) [inline extending/javascript-client/index.md]: good point, will fix.

File: extending/javascript-client/index.md (hunk 2 above)
Comment (adamsilverstein) [inline extending/javascript-client/index.md]: yes, typo. thanks; will fix.

File: extending/javascript-client/index.md (hunk 5 above)
Comment (adamsilverstein) [inline extending/javascript-client/index.md]: ok, will clarify

File: extending/javascript-client/index.md (hunk 6 above)
Comment (adamsilverstein) [inline extending/javascript-client/index.md]: egad. fixing sir! thanks for nit picking.

File: extending/javascript-client/index.md (hunk 8 above)
Comment (adamsilverstein) [inline extending/javascript-client/index.md]: yea, session cache. i'll make that clearer.

**Comment by adamsilverstein** [comment]
@kadamwhite thanks for reviewing and for the feedback!

**Comment by adamsilverstein** [comment]
I made all these corrections ^^. Going to merge this now so we have the Backbone client docs in place. I will update these as things change or new features are added.

**Comment by adamsilverstein** [comment]
props @kadamwhite for reviewing and feedback

**Pull request merged**